crates/project/src/worktree_store.rs

```diff
@@ -302,7 +314,10 @@ impl WorktreeStore {
     }
 
     pub fn add(&mut self, worktree: &Model<Worktree>, cx: &mut ModelContext<Self>) {
```

Comment:
Please add `#[track_caller]` here to see what caused the issue in case the assert fires